### PR TITLE
Port Partioned class from cactoos

### LINKED ISSUE #202: Port class Partitioned
### Expected Behavior
We have this class: https://github.com/yegor256/cactoos/blob/master/src/main/java/org/cactoos/iterable/Partitioned.java

Hint: 

- You must implement the Partitioned Iterator from cactoos, too
- Collections.unmodifiableList is ReadOnlyCollection<T> in C#

### PR DESCRIPTION
**Pull request by lwuLok**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)
closes #202 

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

**Comment by Meerownymous** [comment]
@lwuLok Are you still working in this branch?

**Comment by Meerownymous** [comment]
@lwuLok ping... still working on this?

**Comment by Meerownymous** [comment]
ping...

File: src/Yaapii.Atoms/Enumerator/Partitioned.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                    throw new InvalidOperationException();
+                }
+
+                return buffer;
+            }
+        }
+
+        object IEnumerator.Current => this.Current;
+
+        public void Dispose()
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Enumerator/Partitioned.cs]: Must be removed

File: src/Yaapii.Atoms/Enumerator/Partitioned.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        public void Reset()
+        {
+            this.enumerator.Reset();
+            this.buffer.Clear();
+        }
+
+        private bool Partitionate()
+        {
+            bool result = true;
+
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Enumerator/Partitioned.cs]: remove line

File: src/Yaapii.Atoms/Enumerator/Partitioned.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            this.buffer.Clear();
+
+            if (!this.enumerator.MoveNext())
+            {
+                result = false;
+            }
+
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Enumerator/Partitioned.cs]: remove line

File: src/Yaapii.Atoms/Enumerator/Partitioned.cs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            this.enumerator.Reset();
+            this.buffer.Clear();
+        }
+
+        private bool Partitionate()
+        {
+            bool result = true;
+
+            this.buffer.Clear();
+
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Enumerator/Partitioned.cs]: remove line

File: src/Yaapii.Atoms/Enumerator/Partitioned.cs [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+            else
+            {
+                this.buffer.Add(this.enumerator.Current);
+
+                for (int i = 0; i < this.size - 1 && this.enumerator.MoveNext(); ++i)
+                {
+                    this.buffer.Add(this.enumerator.Current);
+                }
+            }
+
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Enumerator/Partitioned.cs]: remove line

File: tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+namespace Yaapii.Atoms.Enumerator
+{
+    public class PartitionedTests
+    {
+        [Fact]
+        public void CountsPartititions()
+        {
+            int size = 3;
+            IEnumerator<int> enumerator = new List<int>() {1,2,3,4,5,6,7,8,9,10}.GetEnumerator();
+
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs]: remove lines

File: tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            partition.MoveNext();
+            partition.MoveNext();
+            partition.MoveNext();
+
+            Assert.Equal(new List<int>() { 10}, partition.Current);
+        }
+
+
+        [Fact]
+        public void Partionate()
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs]: Partitionates

File: tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        {
+            int size = 1;
+            IEnumerator<string> enumerator = new List<string>() { "Christian","Sauer"}.GetEnumerator();
+
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs]: remove line

File: tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+            var partition = new Partitioned<string>(size, enumerator);
+
+            partition.MoveNext();
+
+           // Assert.Equal(new List<string>() { "Christian", "Sauer" }, partition.Current);
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs]: remove unnecessary comment

File: tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+        [Fact]
+        public void Partionate()
+        {
+            int size = 1;
+            IEnumerator<string> enumerator = new List<string>() { "Christian","Sauer"}.GetEnumerator();
+
+
+            var partition = new Partitioned<string>(size, enumerator);
+
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs]: remove line

File: tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        [Fact]
+        public void ThrowsExceptionOnInvalidSize()
+        {
+            int size = 0;
+            IEnumerator<string> enumerator = new List<string>() { }.GetEnumerator();
+                     
+            Assert.Throws<ArgumentException>(()=>
+            new Partitioned<string>(size, enumerator).MoveNext()
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs]: indentation

File: tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        [Fact]
+        public void CurrentThrowsException()
+        {
+            int size = 2;
+            IEnumerator<string> enumerator = new List<string>() { }.GetEnumerator();
+
+            Assert.Throws<InvalidOperationException>(() =>
+            new Partitioned<string>(size, enumerator).Current
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerator/PartitionedTests.cs]: indentation

**Comment by Meerownymous** [comment]
@lwuLok Pöng

File: src/Yaapii.Atoms/Enumerator/Partitioned.cs (hunk 1 above)
Comment (lwuLok) [reply src/Yaapii.Atoms/Enumerator/Partitioned.cs]: How do I remove them without breaing the interface?

**Pull request merged**